File: applets/src/org/rr0/ufoathome/model/ufo/WitnessModel.java
package org.rr0.ufoathome.model.ufo;

import java.util.Date;
import java.util.GregorianCalendar;

/**
 * @author Jerome Beau
 * @version 0.3
 */
public class WitnessModel {
    private String lastName;
    private String firstName;
    private String email;
    private String address;
    private String town;
    private String zipCode;
    private String country;
    private String phoneNumber;
    private GregorianCalendar birthDate;

    public String getTown() {
        return town;
    }

    public void setTown(String town) {
        this.town = town;
    }

    public GregorianCalendar getBirthDate() {
        if (birthDate == null) {
            birthDate = (GregorianCalendar) GregorianCalendar.getInstance();
        }
        return birthDate;
    }

    public void setBirthDate(Date birthDate) {
        this.birthDate.setTime(birthDate);
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public void setZipCode(String zipCode) {
        this.zipCode = zipCode;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public String getLastName() {
        return lastName;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getEmail() {
        return email;
    }

    public String getAddress() {
        return address;
    }

    public String getZipCode() {
        return zipCode;
    }

    public String getCountry() {
        return country;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public String toString() {
        StringBuffer uudfBuffer = new StringBuffer();
        uudfBuffer.append("<witness");
        uudfBuffer.append(" firstname=\"").append(firstName).append("\"");
        uudfBuffer.append(" lastname=\"").append(lastName).append("\"");
        uudfBuffer.append(" email=\"").append(email).append("\"");
        uudfBuffer.append(" address=\"").append(address).append("\"");
        uudfBuffer.append(" zipcode=\"").append(zipCode).append("\"");
        uudfBuffer.append(" phone-number=\"").append(phoneNumber).append("\"");
        uudfBuffer.append(" birthdate=\"").append(birthDate).append("\"");
        uudfBuffer.append("/>");
        uudfBuffer.append("\n");
        return uudfBuffer.toString();
    }
}

File: applets/src/org/rr0/ufoathome/model/ufo/WitnessModel.ts
export class WitnessModel {
  private lastName: String;
  private firstName: String;
  private email: String;
  private address: String;
  private town: String;
  private zipCode: String;
  private country: String;
  private phoneNumber: String;
  private birthDate: GregorianCalendar;

  public getTown(): String {
    return this.town;
  }

  public setTown(town: String) {
    this.town = town;
  }

  public getBirthDate(): GregorianCalendar {
    if (this.birthDate == null) {
      this.birthDate = (GregorianCalendar);
      GregorianCalendar.getInstance();
    }
    return this.birthDate;
  }

  public setBirthDate(birthDate: Date) {
    this.birthDate.setTime(birthDate);
  }

  public setLastName(lastName: String) {
    this.lastName = lastName;
  }

  public setFirstName(firstName: String) {
    this.firstName = firstName;
  }

  public setEmail(email: String) {
    this.email = email;
  }

  public setAddress(address: String) {
    this.address = address;
  }

  public setZipCode(zipCode: String) {
    this.zipCode = zipCode;
  }

  public setCountry(country: String) {
    this.country = country;
  }

  public setPhoneNumber(phoneNumber: String) {
    this.phoneNumber = phoneNumber;
  }

  public getLastName(): String {
    return this.lastName;
  }

  public getFirstName(): String {
    return this.firstName;
  }

  public getEmail(): String {
    return this.email;
  }

  public getAddress(): String {
    return this.address;
  }

  public getZipCode(): String {
    return this.zipCode;
  }

  public getCountry(): String {
    return this.country;
  }

  public getPhoneNumber(): String {
    return this.phoneNumber;
  }

  public toString(): String {
    const uudfBuffer = new StringBuffer();
    uudfBuffer.append("<witness");
    uudfBuffer.append(" firstname=\"").append(this.firstName).append("\"");
    uudfBuffer.append(" lastname=\"").append(this.lastName).append("\"");
    uudfBuffer.append(" email=\"").append(this.email).append("\"");
    uudfBuffer.append(" address=\"").append(this.address).append("\"");
    uudfBuffer.append(" zipcode=\"").append(this.zipCode).append("\"");
    uudfBuffer.append(" phone-number=\"").append(this.phoneNumber).append("\"");
    uudfBuffer.append(" birthdate=\"").append(this.birthDate).append("\"");
    uudfBuffer.append("/>");
    uudfBuffer.append("\n");
    return uudfBuffer.toString();
  }
}
